plugin/parser/differ/mysql/differ.go

```diff
@@ -122,22 +122,41 @@ func (*SchemaDiffer) SchemaDiff(oldStmt, newStmt string) (string, error) {
 			var alterTableInplaceDropConstraintSpecs []*ast.AlterTableSpec
 
 			oldColumnMap := buildColumnMap(oldNodes, newStmt.Table.Name)
-			for _, columnDef := range newStmt.Cols {
+			oldColumnPositionMap := buildColumnPositionMap(oldStmt)
+			for idx, columnDef := range newStmt.Cols {
 				newColumnName := columnDef.Name.Name.O
 				oldColumnDef, ok := oldColumnMap[newColumnName]
 				if !ok {
+					columnPosition := &ast.ColumnPosition{Tp: ast.ColumnPositionFirst}
+					if idx >= 1 {
+						columnPosition.Tp = ast.ColumnPositionAfter
+						columnPosition.RelativeColumn = &ast.ColumnName{Name: model.NewCIStr(newStmt.Cols[idx-1].Name.Name.O)}
+					}
 					alterTableAddColumnSpecs = append(alterTableAddColumnSpecs, &ast.AlterTableSpec{
 						Tp:         ast.AlterTableAddColumns,
 						NewColumns: []*ast.ColumnDef{columnDef},
+						Position:   columnPosition,
 					})
 					continue
 				}
-				// Compare the two column definitions.
-				if !isColumnEqual(oldColumnDef, columnDef) {
+
+				// Compare the column possitions.
+				columnPosition := &ast.ColumnPosition{Tp: ast.ColumnPositionNone}
+				columnPosInOld := oldColumnPositionMap[newColumnName]
+				if hasColumnsIntersection(oldStmt.Cols[:columnPosInOld], newStmt.Cols[idx+1:]) {
```

Comment:
Is it possible that idx > len(newStmt.Cols)?

Reply:
No, the idx comes from `for idx, columnDef := range newStmt.Cols`